companion/src/shared/timeredit.cpp

```diff
@@ -0,0 +1,281 @@
+
+  div_t qrM = div(abs(seconds), 60);
+  div_t qrH = div(qrM.quot, 60);
+
+  QString val = QString("%1").arg(qrH.rem, 2, 10, QChar('0'));
+
+  if (m_hourDigits)
+    val.prepend(QString("%1:").arg(qrH.quot, m_hourDigits, 10, QChar('0')));
+
+  if (m_showSeconds)
+    val.append(QString(":%1").arg(qrM.rem, 2, 10, QChar('0')));
+
+  if (m_minTime < 0) {
+    if (seconds < 0)
+      val.prepend("-");
+    else
+      val.prepend(" ");
+  }
+  //qDebug() << seconds << qrH.quot << qrH.rem << qrM.quot << qrM.rem << val << text();
+
+  if (text().compare(val)) {
+    setText(val);
+  }
+}
+
+void TimerEdit::incrDecr(int seconds)
+{
+  if (!seconds)
+    return;
+
+  const int cs = timeInSeconds();
+  int s = seconds + cs;
+  if (s < m_minTime)
+    s = m_minTime;
+  if (s > m_maxTime)
+    s = m_maxTime;
+  if (s != cs) {
+    setTime(s);
+    emitValueChanged();
+  }
+}
+
+void TimerEdit::setTimeRange(int minSeconds, int maxSeconds)
+{
+  if (minSeconds <= maxSeconds && (m_minTime != minSeconds || m_maxTime != maxSeconds)) {
+    int digits = 0;
+    if (maxSeconds >= 3600)
+      digits = floorf(logf(round(maxSeconds / 3600)) / logf(10.0f)) + 1;
+
+    bool mod = (m_hourDigits != digits || std::signbit((float)m_minTime) != std::signbit((float)minSeconds));
+
+    m_minTime = minSeconds;
+    m_maxTime = maxSeconds;
+    m_hourDigits = digits;
+
+    if (mod)
+      setupFormat();
+  }
+}
+
+void TimerEdit::setMinimumTime(int minSeconds)
+{
+  setTimeRange(minSeconds, (m_maxTime > minSeconds ? m_maxTime : minSeconds));
+}
+
+void TimerEdit::setMaximumTime(int maxSeconds)
+{
+  setTimeRange((m_minTime < maxSeconds ? m_minTime : maxSeconds), maxSeconds);
+}
+
+void TimerEdit::setShowSeconds(bool showSeconds)
+{
+  if (m_showSeconds != showSeconds) {
+    m_showSeconds = showSeconds;
+    setupFormat();
+  }
+}
+
+
+//// Protected ////
+
+void TimerEdit::textEditedHandler()
+{
+  int sec = timeInSeconds();
+  if (sec < m_minTime)
+    setTime(m_minTime);
+  else if (sec > m_maxTime)
+    setTime(m_maxTime);
+  else
+    return;
+
+  emitValueChanged();
+
+  //qDebug() << text() << sec << getTimeSeconds() << validator()->validate(text(), sec);
+}
+
+void TimerEdit::setupFormat()
+{
+  QString inputRe  = "(?<mins>[0-5][0-9])";
+  QString inputMsk = "99";
+  QString suffx = "\\" % tr("m") % "\\" % tr("m");
+
+  if (m_hourDigits) {
+    inputRe.prepend("(?<hrs>[0-9]*[0-9]):");
+    inputMsk.prepend(":");
+    suffx.prepend(":");
+    for (int i=0; i < m_hourDigits; ++i) {
+      inputMsk.prepend("9");
+      suffx.prepend("\\" % tr("h"));
+    }
+  }
+  if (m_showSeconds) {
+    inputRe.append(":(?<secs>[0-5][0-9])");
+    inputMsk.append(":99");
+    suffx.append(":\\" % tr("s") % "\\" % tr("s"));
+  }
+  if (m_minTime < 0) {
+    inputRe.prepend("(?<pol>-|\\+|\\s)?");
+    inputMsk.prepend("#");
+  }
+
+  inputRe.prepend("^");
+  inputRe.append("\\s?\\[[^\\]]+\\]$");
+  inputMsk.append(" \\[" % suffx % "\\]");
+  //qDebug() << inputMsk << inputRe;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this block of commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3ACommentedCode)